tests/filecheck/parser-printer/duplicate_attribute_keys.mlir

```diff
@@ -0,0 +1,9 @@
+// RUN: xdsl-opt %s --parsing-diagnostics --split-input-file | filecheck %s
+
+"test.op"() {key1, key1} : () -> ()
+// CHECK: Duplicate key 'key1' in dictionary attribute
```

Comment:
Nit: It would be nice if we could have the error be "Duplicate key 'key1' in attributes dictionary" though I appreciate this makes the code more complicated

Reply:
MLIR doesn't do that and I'd rather keep it simple if you don't mind as printing the key name makes the location obvious.

Reply:
fair enough